Content.Shared/Damage/Systems/DamageOnInteractSystem.cs

```diff
@@ -0,0 +1,78 @@
+{
+    [Dependency] private readonly ISharedAdminLogManager _adminLogger = default!;
+    [Dependency] private readonly IGameTiming _gameTiming = default!;
+    [Dependency] private readonly INetManager _net = default!;
+    [Dependency] private readonly DamageableSystem _damageableSystem = default!;
+    [Dependency] private readonly SharedAudioSystem _audioSystem = default!;
+    [Dependency] private readonly SharedPopupSystem _popupSystem = default!;
+    [Dependency] private readonly InventorySystem _inventorySystem = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<DamageOnInteractComponent, InteractHandEvent>(OnHandInteract);
+    }
+
+    /// <summary>
+    /// Damages the user that interacts with the entity with an empty hand and
+    /// plays a sound or pops up text in response. If the user does not have
+    /// proper protection, the user will only be damaged and other interactions
+    /// will be cancelled.
+    /// </summary>
+    /// <param name="entity">The entity being interacted with</param>
+    /// <param name="args">Contains the user that interacted with the entity</param>
+    private void OnHandInteract(Entity<DamageOnInteractComponent> entity, ref InteractHandEvent args)
+    {
+        var totalDamage = entity.Comp.Damage;
+
+        if (!entity.Comp.IgnoreResistances)
+        {
+            // try to get damage on interact protection from either the inventory slots of the entity
+            _inventorySystem.TryGetInventoryEntity<DamageOnInteractProtectionComponent>(args.User, out var protectiveEntity);
+
+            // or checking the entity for  the comp itself if the inventory didn't work
+            if (protectiveEntity.Comp == null && TryComp<DamageOnInteractProtectionComponent>(args.User, out var protectiveComp))
+            {
+                protectiveEntity = (args.User, protectiveComp);
+            }
+
+            // if protectiveComp isn't null after all that, it means the user has protection,
+            // so let's calculate how much they resist
+            if (protectiveEntity.Comp != null)
+            {
+                totalDamage -= protectiveEntity.Comp.DamageProtection;
+                totalDamage.ClampMin(0); // don't let them heal just because they have enough protection
+            }
+        }
+
+        totalDamage = _damageableSystem.TryChangeDamage(args.User, totalDamage, entity.Comp.IgnoreResistances, origin: args.Target);
+
+        if (totalDamage != null && totalDamage.AnyPositive())
+        {
+            _adminLogger.Add(LogType.Damaged, $"{ToPrettyString(args.User):user} injured their hand by interacting with {ToPrettyString(args.Target):target} and received {totalDamage.GetTotal():damage} damage");
+            args.Handled = true;
+
+            if (!_gameTiming.IsFirstTimePredicted || _net.IsServer)
+                return;
+
+            _audioSystem.PlayPvs(entity.Comp.InteractSound, args.Target);
```

Comment:
PlayPredicted?

Reply:
I did try that, it oddly didn't work, I can always try again?

Reply:
Oh, I suppose the isServer check above takes care of it